### ISSUE TITLE
@JsonUnwrappred is ignored by Schema generation

### ISSUE THREAD

**Issue by StepanLeybo** [springfox/springfox#1740]
This issue is a copy of https://github.com/springfox/springfox/issues/192

`@JsonUnwrappred` jackson annotation is ignored by schema generation.

springfox version 2.6.1

```
@RequestMapping(value = "1740", method = RequestMethod.GET)
public Bug1740 bugNNN() {
  return new Bug1740();
}

public class Bug1740 {
  private String value;

  @JsonUnwrapped
  private Bug1740Inner inner;

  public String getValue() {
    return value;
  }

  public Bug1740Inner getInner() {
    return inner;
  }

  private final class Bug1740Inner {
    private String innerValue;

    public String getInnerValue() {
      return innerValue;
    }
  }
}
```
Actual schema generated:
```
"Bug1740": {
  "type": "object",
  "properties": {
    "inner": {
      "$ref": "#/definitions/Bug1740Inner"
    },
    "value": {
      "type": "string"
    }
  }
},
"Bug1740Inner": {
  "type": "object",
  "properties": {
    "innerValue": {
      "type": "string"
    }
  }
}
```
Expected schema:
```
"Bug1740": {
  "type": "object",
  "properties": {
    "value": {
      "type": "string"
    },
    "innerValue": {
      "type": "string"
    }
  }
}
```

PR with test https://github.com/springfox/springfox/pull/1741

**Comment by dilipkrish** [comment]
This really depends on your object mapper configuration and what jackson thinks is your "primary member". In this case it is the `inner` getter (because serialization). As a work around you could annotate the getter with `JsonUnwrapped`. I agree this could be better.

**Comment by dilipkrish** [comment]
Thanks for the PR and a detailed explanation with test cases and examples! Would be awesome if you could help with a PR as well.

**Comment by troybartlett** [comment]
Would love to get this fix in a release... any idea when that might happen?

**Comment by dilipkrish** [comment]
Hopefully end of this month. Still have [a couple of issues](https://github.com/springfox/springfox/issues?q=is%3Aopen+is%3Aissue+label%3A%22in+progress%22) 😫

**Comment by troybartlett** [comment]
@dilipkrish Awesome!

**Comment by evgenycherny** [comment]
+1. Considering heavy use of HATEAOS and need to use JsonUnwrapped in this context, this fix will be AWESOME! Thank you in advance, will try "annotated getter" workaround in the meanwhile.

**Comment by adamsiemion** [comment]
Any plans to fix this? This is a blocker for us.

**Comment by dilipkrish** [comment]
@adamsiemion this has been fixed in 2.9.0

@evgenycherny there is already better support for spring HATEOAS in 2.9.0.

**Comment by davidbilge** [comment]
Can anyone confirm that this is actually fixed? I am on 2.9.2 and `@JsonUnwrapped` is still only considered if the annotation is placed on the getter. If I annotate a field, it does not appear.
